src/components/mapeditor.vue

```diff
@@ -0,0 +1,216 @@
+            class="uk-checkbox"
+            type="checkbox"
+          >
+          &ensp;Show zoom control
+        </label>
+      </div> <!-- uk-margin -->
+
+      <div class="uk-margin">
+        <label class="uk-form-label" for="data-show-legend">
+          <input
+            id="data-show-legend"
+            v-model="showLegend"
+            class="uk-checkbox"
+            type="checkbox"
+          >
+          &ensp;Show legend
+        </label>
+      </div> <!-- uk-margin -->
+
+      <fieldset class="uk-fieldset">
+        <legend class="uk-legend">Address Search</legend>
+        <div class="uk-margin">
+          <label class="uk-form-label" for="data-search-for-address">
+            <input
+              id="data-search-for-address"
+              v-model="searchForAddress"
+              class="uk-checkbox"
+              type="checkbox"
+            >
+            &ensp;Search for an address
+          </label>
+        </div> <!-- uk-margin -->
+
+        <div class="uk-margin">
+          <label class="uk-form-label" for="data-zoom-to-address">
+            <input
+              id="data-zoom-to-address"
+              v-model="zoomToAddress"
+              class="uk-checkbox"
+              type="checkbox"
+              :disabled="!searchForAddress"
+            >
+            &ensp;Zoom to address
+          </label>
+        </div> <!-- uk-margin -->
+
+        <div class="uk-margin">
+          <label class="uk-form-label" for="data-placeholder-text">Placeholder text</label>
+          <div class="uk-form-controls">
+            <input
+              v-model="placeholderText"
+              class="uk-input"
+              id="data-placeholder-text"
+              :disabled="!searchForAddress"
+            >
+          </div>
+        </div> <!-- uk-margin -->
+
+        <div class="uk-margin">
+          <label class="uk-form-label" for="data-show-layer">Show data from layer</label>
+          <div class="uk-form-controls">
+            <select
+              v-model="showDataLayer"
+              class="uk-select"
+              id="data-show-layer"
+              :disabled="!searchForAddress"
+            >
+              <option disabled value="">Please select one</option>
+              <option :value="key" :key="key" v-for="(val, key) in dataLayers">{{ val }}</option>
+            </select>
+          </div>
+        </div> <!-- uk-margin -->
+      </fieldset>
+
+      <div class="uk-panel">
+        <div class="uk-float-right">
+          <button
+            class="uk-button uk-button-default"
+            type="button"
+            @click="onCancel">Cancel</button>
+          <button
+            class="uk-button uk-button-primary"
+            type="button"
+            @click="onSave">Save</button>
+        </div>
+      </div>
+    </form>
+  </div>
+</template>
+
+<script>
+import nanoid from 'nanoid'
+
+export default {
+
+  name: 'map-editor',
+  props: {
+    uid: {
```

Comment:
Do these all have to be props, rather than just passing in a map object from the store?

Reply:
Honestly, I'm not sure anymore. The addition of the store was done late in the process. So these could be vestigial, or they could be necessary.

I think it is worth looking into as another ticket. I don't want it to be a rat hole that holds things up.

Reply:
TBH I find the redux/vuex model intellectually interesting but in practice it adds complication and makes it a lot harder to see the behavior of your application.

I would recommend as a later change to switch to using model objects for the visualization and layers and rely on Vue’s built-in reactivity to observe changes to them.